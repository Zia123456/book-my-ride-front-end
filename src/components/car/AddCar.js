import React, { useState } from 'react';
import { useDispatch } from 'react-redux';
import { addCar } from '../../redux/car/carActions';
import '../../assets/stylesheets/Car.css';

const AddCar = () => {
  const dispatch = useDispatch();
  const [make, setMake] = useState('');
  const [model, setModel] = useState('');
  const [year, setYear] = useState('');
  const [image, setImage] = useState('');
  const [description, setDescription] = useState('');
  const [dailyRate, setDailyRate] = useState('');

  const handleSubmit = (e) => {
    e.preventDefault();
    const car = {
      make,
      model,
      year,
      daily_rate: parseInt(dailyRate, 10),
      image,
      description,
    };
    dispatch(addCar(car));
  };

  return (
    <div className="addForm">
      <div className="innercontainer">
        <div className="add-heading">
          <h2>Add a new car</h2>
        </div>
        <form onSubmit={handleSubmit} className="add-form">
          <input
            className="addInput"
            type="text"
            id="make"
            value={make}
            onChange={(e) => setMake(e.target.value)}
            placeholder="Make"
          />

          <input
            className="addInput"
            type="text"
            id="model"
            value={model}
            onChange={(e) => setModel(e.target.value)}
            placeholder="Model"
          />

          <input
            className="addInput"
            type="text"
            id="year"
            value={year}
            onChange={(e) => setYear(e.target.value)}
            placeholder="Year"
          />

          <input
            className="addInput"
            type="text"
            id="image"
            value={image}
            onChange={(e) => setImage(e.target.value)}
            placeholder="Image-url"
          />

          <textarea
            className="addInput"
            id="description"
            value={description}
            onChange={(e) => setDescription(e.target.value)}
            placeholder="Descriptions"
          />

          <input
            className="addInput"
            type="number"
            id="daily-rate"
            value={dailyRate}
            onChange={(e) => setDailyRate(e.target.value)}
            placeholder="Daily Rate"
          />

          <button type="submit" className="btn-add-car">
            Add Car
          </button>
        </form>
      </div>
    </div>
  );
};

export default AddCar;
