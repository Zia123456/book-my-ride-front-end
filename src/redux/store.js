import { configureStore } from '@reduxjs/toolkit';
import authReducer from './auth/authSlice';
import dataReducer from './reservation/reservationsListSlice';
import carReducer from './car/carSlice';
import addReservationReducer from './reservation/addReservationSlice';

const store = configureStore({
  reducer: {
    data: dataReducer,
    auth: authReducer,
    cars: carReducer,
    addReservation: addReservationReducer,
  },
});

export default store;
