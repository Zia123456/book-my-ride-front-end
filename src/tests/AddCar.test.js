import React from 'react';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../redux/store';
import AddCar from '../components/car/AddCar';

it('Add car component should render correctly', () => {
  const result = render(
    <React.StrictMode>
      <Router>
        <Provider store={store}>
          <AddCar />
        </Provider>
      </Router>
    </React.StrictMode>,
  );
  expect(result).toMatchSnapshot();
});
