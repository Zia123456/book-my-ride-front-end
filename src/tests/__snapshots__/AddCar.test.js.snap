// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add car component should render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="addForm"
      >
        <div
          class="innercontainer"
        >
          <div
            class="add-heading"
          >
            <h2>
              Add a new car
            </h2>
          </div>
          <form
            class="add-form"
          >
            <input
              class="addInput"
              id="make"
              placeholder="Make"
              type="text"
              value=""
            />
            <input
              class="addInput"
              id="model"
              placeholder="Model"
              type="text"
              value=""
            />
            <input
              class="addInput"
              id="year"
              placeholder="Year"
              type="text"
              value=""
            />
            <input
              class="addInput"
              id="image"
              placeholder="Image-url"
              type="text"
              value=""
            />
            <textarea
              class="addInput"
              id="description"
              placeholder="Descriptions"
            />
            <input
              class="addInput"
              id="daily-rate"
              placeholder="Daily Rate"
              type="number"
              value=""
            />
            <button
              class="btn-add-car"
              type="submit"
            >
              Add Car
            </button>
          </form>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="addForm"
    >
      <div
        class="innercontainer"
      >
        <div
          class="add-heading"
        >
          <h2>
            Add a new car
          </h2>
        </div>
        <form
          class="add-form"
        >
          <input
            class="addInput"
            id="make"
            placeholder="Make"
            type="text"
            value=""
          />
          <input
            class="addInput"
            id="model"
            placeholder="Model"
            type="text"
            value=""
          />
          <input
            class="addInput"
            id="year"
            placeholder="Year"
            type="text"
            value=""
          />
          <input
            class="addInput"
            id="image"
            placeholder="Image-url"
            type="text"
            value=""
          />
          <textarea
            class="addInput"
            id="description"
            placeholder="Descriptions"
          />
          <input
            class="addInput"
            id="daily-rate"
            placeholder="Daily Rate"
            type="number"
            value=""
          />
          <button
            class="btn-add-car"
            type="submit"
          >
            Add Car
          </button>
        </form>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
